src/drivers/code_hosting/gitlab.sh

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+
+function create_pull_request {
+  local repository=$1
+  local branch=$2
+  local parent_branch=$3
+
+  local to_compare
+  if [ "$parent_branch" = "$MAIN_BRANCH_NAME" ]; then
+    # Allow GitLab to redirect to the proper place if this repository is a fork
+    to_compare="$branch"
```

Comment:
I'm not saying the tests wouldn't pass but that is open a sample URL where only one branch is specified on Gitlab, then you get a 404. The tests don't actually open a browser but only check that the command used to open a browser is called with the expected URL. In this case I believe the test have the wrong expected URL. 

Ideally to implement this freature the flow should have been the following: go to a web page on gitlab for creating a new pull request, take note of the URL structure, write tests that expect the URL to match that format, write the code to make it happen (using the github / bitbucket implementations as reference)